x-pack/plugins/observability_solution/infra/server/lib/alerting/common/utils.ts

```diff
@@ -176,8 +176,10 @@ export const getMetricsViewInAppUrlWithSpaceId = ({
   metricsExplorerLocator?: LocatorPublic<MetricsExplorerLocatorParams>;
   additionalContext?: AdditionalContext;
 }) => {
+  const flattenedContext = additionalContext && flattenObject(additionalContext);
+
   const fields = {
-    ...additionalContext,
+    ...flattenedContext,
```

Comment:
we could also use existing `flattenAdditionalContext` function

```suggestion
    ...flattenAdditionalContext(additionalContext),
```